src/generic-container.ts

```diff
@@ -19,20 +29,52 @@ import {
 import { RandomUuid, Uuid } from "./uuid";
 import { HostPortWaitStrategy, WaitStrategy } from "./wait-strategy";
 
-export class GenericContainer implements TestContainer {
-  public static async fromDockerfile(
-    context: BuildContext,
-    uuid: Uuid = new RandomUuid(),
-    dockerClientFactory: DockerClientFactory = new DockerodeClientFactory()
-  ): Promise<GenericContainer> {
-    const image = uuid.nextUuid();
-    const tag = uuid.nextUuid();
+export class GenericContainerBuilder {
+  private factory: DockerClientFactory;
+  private uuid: Uuid;
+  private buildArgs: BuildArgs;
+
+  constructor(private context: BuildContext) {
```

Comment:
The constructor style used elsewhere in the project is more succinct and marks the fields as readonly:

```typescript
constructor(private readonly context: BuildContext, ...) {
}
```